File: tests/fixtures/negative/security/WeakKeySize.java
package test;

class Crypto {
    void keys(int size, Generator other) throws Exception {
        java.security.KeyPairGenerator rsa = java.security.KeyPairGenerator.getInstance("RSA");
        rsa.initialize(2048);
        javax.crypto.KeyGenerator aes = javax.crypto.KeyGenerator.getInstance("AES");
        aes.init(128);
        aes.init(size);
        other.initialize(1024);
    }
}

class Generator {
    void initialize(int size) {}
}

File: tests/fixtures/negative/security/WeakKeySize.kt
package test

import java.security.KeyPairGenerator
import javax.crypto.KeyGenerator

class Crypto {
    fun keys(size: Int, other: Generator) {
        val rsa = KeyPairGenerator.getInstance("RSA")
        rsa.initialize(2048)
        val aes = KeyGenerator.getInstance("AES")
        aes.init(128)
        aes.init(size)
        other.initialize(1024)
    }
}

class Generator {
    fun initialize(size: Int) {}
}
